data/catalogs/test_data_catalog.py

```diff
@@ -0,0 +1,117 @@
+the version of the dataset source by supplying the call with a -dsv flag and version
+number. The dataset is opened and if the dataset contains geo data the dataset is clipped
+to a small bounding box surrounding Deltares Delft office.
+Example dataset source test call:
+    'python test_data_catalog.py path/to/datacatalog.yml -ds chelsa -dsv 1.2
+    -r "{'bbox':[4.333409,51.962159,4.42835,52.006873]}"'
+
+In addition the passed data catalog yaml is checked if it is a valid data catalog yaml.
+
+"""
+import argparse
+import json
+
+from dask.distributed import Client
+
+from hydromt import DataCatalog
+from hydromt.log import setuplog
+from hydromt.validators.data_catalog import DataCatalogValidator
+
+client = Client(processes=False)
+logger = setuplog()
+
+
+def test_dataset(args, datacatalog):
+    """Tests a given dataset on opening the dataset and minimal processing.
+
+    Datasets containing geo data will be clipped to a small bounding box. This bounding box can
+    be given by a user with the -bbox flag.
+    """
+    source = datacatalog.get_source(source=args.dataset, version=args.dataset_version)
+    if args.region:
+        region_json = args.region.replace("'", '"')
+        bbox = json.loads(region_json)["bbox"]
+        print(bbox)
+    else:
+        bbox = [
+            4.333409,
+            51.962159,
+            4.42835,
+            52.006873,
+        ]  # small area surrounding Deltares Delft office
+    logger.info(
+        f"Retrieving {args.dataset} {source.data_type} and clipping by {bbox} bounding box"
+    )
+    if source.data_type == "GeoDataFrame":
+        dataset = datacatalog.get_geodataframe(args.dataset, bbox=bbox)
+    elif source.data_type == "GeoDataset":
+        dataset = datacatalog.get_geodataset(args.dataset, bbox=bbox)
+    elif source.data_type == "RasterDataset":
+        dataset = datacatalog.get_rasterdataset(args.dataset, bbox=bbox)
+    elif source.data_type == "DataFrame":
+        dataset = datacatalog.get_dataframe(args.dataset)
+    elif source.data_type == "Dataset":
+        dataset = datacatalog.get_dataset(args.dataset)
+    assert dataset is not None
+
+
+def test_data_catalog(args, datacatalog):
+    """Tests the paths of the given data catalog."""
+    error_count = 0
+    logger.info("Checking paths of data catalog sources")
+    for source in datacatalog.get_source_names():
+        try:
+            logger.info(f"Checking paths of {source}")
+            datacatalog.get_source(source)._resolve_paths()
+        except FileNotFoundError as e:
+            logger.error(f"File not found for dataset source {source}: {e}")
+
+            error_count += 1
+        except ValueError as e:
+            logger.error(
+                f"Something went wrong with creating path string for dataset source {source}: {e}"
+            )
+            error_count += 1
+    logger.info(f"Encountered {error_count} errors")
```

Comment:
think this should be logger.error if error_count >0

Reply:
changed it logger.error